fsi/status.go

```diff
@@ -73,23 +97,39 @@ func (fsi *FSI) Status(dir string) (changes []StatusItem, err error) {
 		}
 	}
 
-	// stored.DropDerivedValues()
+	stored.DropDerivedValues()
 
 	ds, mapping, err := ReadDir(dir)
 	if err != nil {
 		return nil, err
 	}
-	// ds.DropDerivedValues()
+	ds.DropDerivedValues()
 
 	// if err = validate.Dataset(ds); err != nil {
 	// 	return nil, fmt.Errorf("dataset is invalid: %s" , err)
 	// }
 
+	storedComponents := dsComponents(stored)
+
+	for cmpName := range storedComponents {
+		// when reporting deletes, ignore "bound" components that must/must-not
+		// exist based on external conditions
+		if cmpName != componentNameDataset && cmpName != componentNameStructure && cmpName != componentNameCommit {
```

Comment:
Does this mean that if a user deletes the structure but changes nothing else, then `status` won't report any changes? Is that intentional, or are we only trying to handle the case of deleting both body and structure leading to a single message?

Reply:
yeah, it's true that deleting structure in this case won't produce a change. I think we should handle that scenario in a yet-to-be-written subroutine that checks for values that will be inferred & creates status items for those.